src/utils/renderNamePattern.ts

```diff
@@ -3,10 +3,14 @@ import { extname } from './path';
 import { isPlainPathFragment } from './relativeId';
 
 export function renderNamePattern(
-	pattern: string,
+	pattern: ((name: string) => string) | string,
```

Comment:
I was choosing a name as this was the only one that exists for all and as we return a pattern. 
I will look into it but I have not much hope.
This is using the emitName with extension normally i am not sure why entry points do not have an extension it must be a deeper design consideration.

Reply:
ok going for this now
```js
// entrypoint (chunk)
{
  format: [Function: format],
  hash: [Function: hash],
  name: [Function: name]
}
//emitFile asset Chunk
{
  hash: [Function: hash],
  ext: [Function: ext],
  extname: [Function: extname],
  name: [Function: name]
}

Reply:
I would really prefer if we could instead use the prerendered chunk object and especially not include the hash. The reason for the former is that it contains a list of all modules in the chunk, which I think is very useful, the latter because very soon I will start reworking the hashing algorithm and once it is done, it will no longer be available when chunk names are generated. I can have a look this evening.

Reply:
@lukastaegert 
I think i now did the best do able i did simply supply both so we can get the best of both worlds and the user can choose if he wants the cost of additional getFileInfo() 

i now supply 
```js
{ replacements, getFileInfo<()> }
``` 

the user can choose if he wants to work with the replacements or getFileinfo() function and i supply all asset info in the same way so there is for example a always consistent getFileInfo().name property even for assets